src/ansys/simai/core/utils/misc.py

```diff
@@ -41,3 +48,24 @@ def dict_get(obj: dict, *keys: str, default=None):
     for k in keys:
         obj = obj.get(k, {}) or {}
     return obj or default
+
+
+def warn_package_outdated(package: str, current_version: str, latest_version: str):
+    try:
+        version_current = Version.parse(current_version)
+        version_latest = Version.parse(latest_version)
+    except ValueError as e:
+        logger.debug(f"Could not parse package version: {e}")
+    else:
+        if version_current < version_latest:
+            warn_template = (
+                f"A new version of {package} is %s. "
+                "Upgrade to get new features and ensure compatibility with the server."
+            )
+            if (
+                version_current.major < version_latest.major
+                or version_current.minor < version_latest.minor
```

Comment:
I still don't like that we're raising an error on minor change. I guess it kind of works for now cause we're on 0.x.x so i'll just not bump the minor unless there's big breaking changes. But randomly breaking all the scripts is bad IMO.

Reply:
Agree, cahnged to a "critical" log